package-build.el

```diff
@@ -1080,9 +1094,10 @@ and a cl struct in Emacs HEAD.  This wrapper normalises the results."
 
     (package-build--message "\n;;; %s\n" file-name)
     (let* ((version (package-version-join
-                     (version-to-list
+                     (package-build--version-to-list
```

Comment:
OK, leaving the previous comment but I get why you are doing this here. I think we should drop it based on my comments above. The *checkout* functions should basically get the tag name and check that tag out, but should still return a valid version. I would prefer that the rest of code not ever need to worry about the version regexp.